lib/blockchain/migrations.js

```diff
@@ -207,6 +261,134 @@ class MigrateChainState extends AbstractMigration {
   }
 }
 
+/**
+ * Migrate block and undo data to BlockStore from chainDB.
+ */
+
+class MigrateBlockStore extends AbstractMigration {
+  /**
+   * Create MigrateBlockStore object.
+   * @param {ChainMigratorOptions} options
+   */
+
+  constructor(options) {
+    super(options);
+
+    this.options = options;
+    this.logger = options.logger.context('chain-migration-blockstore');
+    this.db = options.db;
+    this.ldb = options.ldb;
+    this.blocks = options.db.blocks;
+  }
+
+  /**
+   * Check if the ChainDB has the blocks.
+   * @returns {Promise}
+   */
+
+  async check() {
+    if (this.options.spv)
+      return types.FAKE_MIGRATE;
+
+    return types.MIGRATE;
+  }
+
+  /**
+   * Migrate blocks and undo blocks
+   * @param {Batch} b
+   * @returns {Promise}
+   */
+
+  async migrate() {
+    assert(this.blocks, 'Could not find blockstore.');
+
+    this.logger.info('Migrating blocks and undo blocks.');
+    this.logger.info('This may take a few minutes...');
+
+    await this.migrateBlocks();
+    await this.migrateUndoBlocks();
+
+    this.logger.info('Compacting database...');
+    this.logger.info('This may take a few minutes...');
+    await this.ldb.compactRange();
```

Comment:
Are there undesired consequences if the user kills the process while `compactRange` is running?

Reply:
TLDR: It should not create problems.
NOTE: If migration is terminated, state wont increase so on restart that
migration will restart. That's why it's important to write migrations with
restarts in mind (e.g. FileBlockstore doing it in batches)

okay, I am not 100% sure on this one tbh, cause there are many things to
consider. There's not single way to force the process to exit, but we can
discuss KILL and SIGINT.
  bin/node has SIGINT handler, which means that default nodejs handler wont kick
in. It will call `node.close` which will in turn close everything and wait if
some things need time to close. In this scenario compactRange will just finish
executing before everything closes.
  If something else is used to kill the process, then I can not guarantee that
level-db compactRange wont screw things up. I will need to review it deeply to
confidently answer the question. After several tests I could not produce any
corruption, on the next start it just continued doing compacting db. But the
same is true for many other operations, like Batch - what if process is killed
while level-db batch.write is being executed.

Reply:
Ok, sounds like this is the best we can do anyway